Create projectComplete() dependency method in JvmComponentDepenencies (#21830)
gradle/gradle: subprojects/docs/src/docs/release/notes.md

Similarly, the new Gradle property `org.gradle.debug.host` now enables [running the Gradle process with the debugger server](userguide/troubleshooting.html#sec:troubleshooting_build_logic)
accepting connections via network on Java 9+.

#### Introduced `projectComplete()` dependency for creating unit-like test suites

> The [JVM test suite](userguide/jvm_test_suite_plugin.html) `dependencies` block now has support for depending on the complete view of the current project during compile-time. Previously it was only possible to depend on the current project's API. This allows declaration of unit-like test suites which have access to dependencies of project internals which are not declared on the `api` or `compileOnlyApi` configurations. 

Reviewer: > which have access to dependencies of project internals which are not declared on the `api` or `compileOnlyApi` configurations. 

This is hard to parse for me due to the double use of `which`. I can't think of a good improvement though, perhaps @nathan-contino has some ideas.

> #### Introduced `projectComplete()` dependency for creating unit-like test suites
> 
> The [JVM test suite](userguide/jvm_test_suite_plugin.html) `dependencies` block now has support for depending on the complete view of the current project during compile-time. Previously it was only possible to depend on the current project's API. This allows declaration of unit-like test suites which have access to dependencies of project internals which are not declared on the `api` or `compileOnlyApi` configurations. 

Reviewer 2: ```suggestion
#### Introduced `projectComplete()` dependency for test suites with access to project internals

The [JVM test suite](userguide/jvm_test_suite_plugin.html) `dependencies` block now
supports depending on the complete view of the current project during compile-time.
Previously it was only possible to depend on the current project's API. This allows
test suites with access to project internals that are not declared on
the `api` or `compileOnlyApi` configurations.
```
Suggestion: minor rewords, remove "unit-like" since it's an unfamiliar term that we're unlikely to use in the future.
Added explicit newlines for sanity's sake.

> For example, the following snippet uses the new `projectComplete()` API to define a new unit-like test suite:

Reviewer 3: I don't like the name `unit-like` here - what kind of tests are these?  I think you should briefly describe a case where you would want to access the internals and name the suite accordingly.

Reviewer 2: ```suggestion
For example, the following snippet uses the new `projectComplete()` API to define a test suite with access to project internals:
```
Suggestion: use the definition of "unit-like" instead of the term.
